frontend/src/components/SemesterBlock.tsx

```diff
@@ -136,29 +162,27 @@ class SemesterBlockComponent extends React.Component<
             this.props.handleAddClasses(courses, this.props.semester)
           }
         ></AddClassModal>
-
-        <Container>
-          <Droppable droppableId={this.props.semester.termId.toString()}>
-            {provided => (
-              <ClassList
-                innerRef={provided.innerRef as any}
-                {...provided.droppableProps}
-              >
-                {this.renderBody()}
-                {provided.placeholder}
-              </ClassList>
-            )}
-          </Droppable>
-          <AddButtonContainer>
-            <AddButton onClick={this.showModal.bind(this)}></AddButton>
-          </AddButtonContainer>
-        </Container>
+        {!!this.props.warning ? (
+          <Tooltip title={this.props.warning.message} placement="top">
+            {this.renderContainer()}
+          </Tooltip>
+        ) : (
+          this.renderContainer()
+        )}
       </div>
     );
   }
 }
 
 const mapStateToProps = (state: AppState, ownProps: SemesterBlockProps) => ({
+  warning:
+    getWarningsFromState(state).filter(
+      w => w.termId === ownProps.semester.termId
+    ).length > 0
+      ? getWarningsFromState(state).filter(
+          w => w.termId === ownProps.semester.termId
+        )[0]
+      : undefined,
```

Comment:
Rather than just grabbing [0] here, should probably just be the entire filtered array.